### PR TITLE
Adds option to collect call history for all turns - in select command

### PR DESCRIPTION
**Pull request by janaab11**
The main functionality is to return the all the turns of a list of call ids (random or selected). This PR adds a mutator that collect the call history of each turn, and appends it as a new field. 

The CLI option has only been added for the `select` command, since thats where such a utility has been needed (in the past). Can be added for the `sample` command as well

### CODE REVIEW

File: skit_calls/data/mutators.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    turns_df[constants.CALL_HISTORY] = turns_df[constants.CALL_UUID].apply(
+        lambda call_key: get_column_value(
+            calls_df.set_index(constants.CALL_UUID),
+            idx=call_key,
+            column_key=constants.CALL_HISTORY,
+        )
+    )
+
+    # filter call data to only keep history
+    ## assumption here is that turns will be in ascending order of created time
```
Comment (janaab11) [inline skit_calls/data/mutators.py:42]: highlighting this

**Comment by janaab11** [comment]
**What needs to be fixed:**

`Call history of a turn` is basically a list of the previous turns in the call. This means there is an implicit assumption that _the turns are ordered by the time they were created_. A way to enforce this is:

1. to order turns by `conversations.created_at`, after grouping the turns of a call
2. to order turns by `conversations.id`, after grouping the turns of the call

If we follow 1, we would be adding an extra field, that needs to be read as a datetime for ordering.

If we follow 2, we could just replace `conversations.uuid` with `conversations.id`. This field would be both an index for turns and a proxy for `conversations.created_at` - since ordering by either (within a call), results in the same output.

Either way, need some edits to the query being run (that is stored privately on gitlab)

@ltbringer your thoughts ?

**Comment by ltbringer** [comment]
> to order turns by conversations.created_at, after grouping the turns of a call

I was warned about the reliability of the `created_at` field. Due to asynchronous operations, the order is not guaranteed by timestamps. The recommendation was to use the `conversation.id`

File: skit_calls/data/model.py [hunk 2]
```diff
@@ -83,6 +83,7 @@ def get_url(base: MaybeString, path: MaybeString) -> MaybeString:
 class Turn:
     call_id: str = attr.ib(kw_only=True, repr=True, converter=str)
     call_uuid: str = attr.ib(kw_only=True, repr=False)
+    conversation_id: int = attr.ib(kw_only=True, repr=True, converter=int)
```
Comment (janaab11) [inline skit_calls/data/model.py:86]: this might need reviewing

**Comment by janaab11** [comment]
added `conversation_id` field and used it for sorting turns. two things to do, before merging this PR:

- review the attribute definition in the `Turn` class (see comment above)
- merge changes in the private repo - this adds conversations.id as an _additional_ field in the query

@ltbringer assuming you can look at these, since you are the maintainer of the private repo

**Review by ltbringer — APPROVED** [review]

**Pull request merged**